xonsh/history/base.py

```diff
@@ -163,3 +169,35 @@ def clear(self):
         memory.
         """
         pass
+
+    def validate_ignore_regex(self):
+        """Validates self.history_ignore_regex to make sure it's a valid
+        regex. If invalid, sets it to None and outputs a warning."""
+        if self.history_ignore_regex is not None:
+            try:
+                re.compile(self.history_ignore_regex)
+            except re.error:
```

Comment:
I think we can store the compile result lazily. Like a `cached_property`

Reply:
This is just meant to validate the expression, but the point is taken that I can compile it once and use `match` on the compiled expression.